src/v/cloud_storage/recursive_directory_walker.cc

```diff
@@ -50,18 +50,20 @@ recursive_directory_walker::walk(
 
         try {
             ss::file target_dir = co_await open_directory(target);
+            size_t files_in_dir{0};
             auto sub = target_dir.list_directory(
               [&files,
                &current_cache_size,
                &dirlist,
+               &files_in_dir,
                _target{target},
                _tracker{tracker}](ss::directory_entry entry) -> ss::future<> {
                   auto target{_target};
                   auto tracker{_tracker};
-                  vlog(cst_log.debug, "Looking at directory {}", target);
 
                   auto entry_path = std::filesystem::path(target)
                                     / std::filesystem::path(entry.name);
+                  files_in_dir++;
```

Comment:
nit: I think `files_in_dir` could just be a bool, and toggled with `if unlikely(!files_in_dir)`, although admittedly the memory and cpu savings would be tiny.